lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -74,24 +78,46 @@ export default function LMSFilePicker({
         authToken,
         path: listFilesApi.path,
       }));
-      setDialogState({ ...INITIAL_DIALOG_STATE, state: 'fetched', files });
+      const title =
+        files.length === 0 ? 'No files available' : INITIAL_DIALOG_STATE.title;
+      const buttonAction =
+        files.length === 0 ? 'none' : INITIAL_DIALOG_STATE.buttonAction;
+      setDialogState({
+        ...INITIAL_DIALOG_STATE,
+        state: 'fetched',
+        files,
+        title,
+        buttonAction,
+      });
     } catch (e) {
+      const buttonAction = authorizationAttempted
+        ? 'authorize_retry'
+        : 'authorize';
       if (e instanceof ApiError && !e.errorMessage) {
         // If the server returned an error, but provided no details, assume
         // an authorization failure.
-        setDialogState({ ...INITIAL_DIALOG_STATE, state: 'authorizing' });
+        setDialogState({
+          ...INITIAL_DIALOG_STATE,
+          state: 'authorizing',
+          title: 'Allow file access',
+          buttonAction,
+        });
       } else {
         // Otherwise, display the error to the user.
-        setDialogState({ ...INITIAL_DIALOG_STATE, state: 'error', error: e });
+        setDialogState({
+          ...INITIAL_DIALOG_STATE,
+          state: 'error',
+          title: 'Error accessing files',
+          error: e,
+          buttonAction: 'retry',
+        });
       }
     }
-  }, [authToken, listFilesApi]);
+  }, [authToken, listFilesApi, authorizationAttempted]);
 
   // Execute the authorization flow in a popup window and then attempt to
   // fetch files.
   const authorizeAndFetchFiles = useCallback(async () => {
-    setDialogState({ ...INITIAL_DIALOG_STATE, state: 'authorizing' });
-
```

Comment:
I think this is not necessary because the state has already been set by `fetchFiles`.